soda_machine.rb

```diff
@@ -4,15 +4,21 @@ class SodaMachine
   def initialize(args = {})
     @sodas = args[:sodas]
     @cash = args[:cash]
+    
   end
 
   def current_inventory_count
+    @sodas.length
+  
   end
 
   def find_soda(soda_brand)
+    @sodas.each do |soda, price|
```

Comment:
Take a little time to familiarize yourself with the data types that you are dealing with before diving into your code. Here you are iterating over  `@sodas` (an `Array`), but you are telling ruby to expect two elements, `soda` and `price`. Each element in the `@sodas` Array is a `Soda` object which has attributes `brand` and `price`.